### ISSUE TITLE
Effector does not warn, if store without sid is used as a value in fork

### ISSUE THREAD

**Issue by AlexandrHoroshih** [effector/effector#567]
**What is the current behavior:**
Store without sid still can be used as a value in fork, even though it leads to weird behaviour without any warnings or errors

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**

REPL: https://share.effector.dev/35CeFqED

**What is the expected behavior:**
Effector warns or errors, if store without sid is used in fork values, so user can check babel-plugin usage or provide sid's manually

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**
`effector@22.0.0` and up

**Comment by zerobias** [comment]
I revisited your example and found that it passed **an object** with "null" property to `values` which is perfectly valid: "null" string is valid sid identifier and indeed, we are switched to js maps and tuples for exactly that reason. Object with sid keys should be treated as internal feature and shouldn't be used in production code

**Comment by AlexandrHoroshih** [comment]
@zerobias 
My reproduce, probably, was not correct, sorry about that(
I had manually added null sids to reproduce the case from the community chat - where user forgot to add babel-plugin and noticed that there is no values at the client
It was best i came up with at the moment - because `effector/babel-plugin` is always active in the REPL

Better reproduce, not in the effector's repl and without manually set sids or babel-plugin:
https://stackblitz.com/edit/js-yvdya1
It is much closer to actual problem

```ts
const event = createEvent();
const $abc = createStore(0).on(event, (s) => s + 1);

const scope = fork();

allSettled(event, { scope });
// event is called in scope, $abc is changed from 0 to 1

const values = serialize(scope);
console.log(values);
// but there is no { [$abc.sid]: 1 } in serialized data 🤷‍♂️ 

// later at the client
const clientScope = fork({
  values,
});
...
const abc = useStore($abc)
// still returns 0
// even though logic was ran at the server and serialized state is passed to the client without any errors
```
From the user's point of view everything is done in a correct way and according to examples, but it just doesn't work in a confusing way

**Comment by zerobias** [comment]
Some stores in application could exists without sids, for example stores created by `.map`, what should happens with them?

**Comment by zerobias** [comment]
Relying on babel is an issue, big headache, I'm trying not to say "an error", this is an unreliable [obscure](https://github.com/effector/effector/issues/603#issuecomment-1014130787) tool which rapidly fall out from favor due to objective reasons. Could we just tell users to insert sids manually?

**Comment by AlexandrHoroshih** [comment]
> Some stores in application could exists without sids, for example stores created by .map, what should happens with them?

Mapped stores are not serialized anyway, so they are not a subject to this problem 🤔 
If user decides to manually `{ [mappedStore.sid]: value }` - so be it 🤷‍♂️ 

I think, that only case, that should be handled is:
1. a normal store
2. which was "touched" during allSettled
3. does not have `serialize: "ignore"`
4. and does not have a sid

If there is a store like this, then forked scope cannot be serialized in a reliable way, data will be missing

I see that [`serialize` just goes over `scope.sidValuesMap`](https://github.com/effector/effector/blob/e3f356103275021735fe8aa72a998f1fef1038f8/src/effector/fork/serialize.ts#L16)
And if there is a normal store without sid, then it just will not be presented in this Map 🤔 

Maybe effector could mark scope as "non-serializable", if normal store without sid and without "ignore" was changed in this scope?

Then it would be possible to just check for that in `serialize`:
```ts
if (scope.warnSerialize) {
 // not sure, if it should throw
 console.error("[effector] provided scope cannot be serialized in a reliable way because some stores are missing sid. Please, check that babel-plugin is working or provide sid manually")
}
```

**Comment by zerobias** [comment]
Seem reasonable, but how it will work with packages of the ecosystem? Would it be enough for them to declare all their internal stores as unserializable?

**Comment by AlexandrHoroshih** [comment]
> Would it be enough for them to declare all their internal stores as unserializable?

Yep, i think `serialize: "ignore"` will be enough for internal stores.
I guess, if store value must be transferred over the network, then library developer always can use babel-plugin or even hardcode sids manually like `sid: "lib-name/store-name"`
